fast/stages/3-project-factory/dev/main.tf

```diff
@@ -17,106 +17,12 @@
-        for role, members in lookup(v, "iam_additive", {}) :
-        role => [
-          for member in members :
-          endswith(member, "@") ? "${member}${var.organization.domain}" : member
-        ]
-      }
-      group_iam = {
-        for role, groups in lookup(v, "group_iam", {}) :
-        role => [
-          for group in groups :
-          endswith(group, "@") ? "group:${group}${var.organization.domain}" : "group:${group}"
-        ]
-      }
-      org_policies = lookup(v, "org_policies", null)
-      tag_bindings = {
-        for k2, v2 in lookup(v, "tag_bindings", {}) :
-        k2 => lookup(var.tag_values, v2, v2)
-      }
-      # management
-      services = distinct(concat(lookup(v, "services", []), local._default_services))
-      contacts = { for x in lookup(v, "contacts", []) : x => ["ALL"] }
-      # billing budget
-      budget = lookup(v, "budget", null)
-      # networking
-      shared_vpc_service_config = lookup(v, "vpc", null) == null ? null : {
-        host_project         = lookup(local._all_vpc_hosts, v.vpc.host_project, v.vpc.host_project)
-        service_identity_iam = lookup(v.vpc, "service_identity_iam", null)
-      }
-    })
-  }
-  subnet_bindings = flatten([
-    for k, v in local._projects_raw : [
-      for subnet, members in try(v.vpc.subnets_iam, {}) : [
-        for member in members : {
-          host_project = lookup(local._all_vpc_hosts, v.vpc.host_project, v.vpc.host_project),
-          region       = split("/", subnet)[0],
-          subnet       = split("/", subnet)[1],
-          member       = endswith(member, "@") ? "${member}${var.organization.domain}" : member
-        }
-      ]
-    ]
-  ])
-}
-
-module "projects" {
-  source                    = "../../../../modules/project"
-  for_each                  = local.projects
-  billing_account           = each.value.billing_account
-  parent                    = each.value.parent
-  prefix                    = var.prefix
-  name                      = each.value.name
-  descriptive_name          = each.value.descriptive_name
-  labels                    = each.value.labels
-  services                  = each.value.services
-  iam                       = each.value.iam
-  iam_additive              = each.value.iam_additive
-  group_iam                 = each.value.group_iam
-  org_policies              = each.value.org_policies
-  contacts                  = each.value.contacts
-  shared_vpc_service_config = each.value.shared_vpc_service_config
-  tag_bindings              = each.value.tag_bindings
-}
-
-resource "google_compute_subnetwork_iam_member" "default" {
-  for_each   = { for b in local.subnet_bindings : join("/", values(b)) => b }
-  project    = each.value.host_project
-  subnetwork = "projects/${each.value.host_project}/regions/${each.value.region}/subnetworks/${each.value.subnet}"
-  region     = each.value.region
-  role       = "roles/compute.networkUser"
-  member     = each.value.member
+  hierarchy_files = try(
```

Comment:
these `try()` will hide filesystem errors (e.g. permission issues, non-existing paths, etc). Can we do an old-style ternary?

Reply:
Good point